Задача 1. 

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [1]:
import numpy as np
p_100 = (np.math.factorial(100)/(np.math.factorial(85)*np.math.factorial(15)))*0.8**85*0.2**15
p_100

0.048061793700746556

Ответ: 4,8%

Задача 2.

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.
Какова вероятность, что ни одна из них не перегорит в первый день?
Какова вероятность, что перегорят ровно две?

In [2]:
lam  = 5000*0.0004
p_0 = np.exp(lam*(-1))
p_0

0.1353352832366127

Ответ: вероятность, что ни одна из них не перегорит в первый день - 13,53%

In [3]:
p_2 = (lam**2/(np.math.factorial(2)))*np.exp(lam*(-1))
p_2

0.2706705664732254

Ответ: вероятность, что перегорят ровно две - 27,06%

Задача 3.

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [4]:
p_144_70 = (np.math.factorial(144)/(np.math.factorial(70)*np.math.factorial(74)))*0.5**70*0.5**74
p_144_70

0.06281178035144776

Ответ: 6,28%

Задача 4.

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.
Какова вероятность того, что все мячи белые?
Какова вероятность того, что ровно два мяча белые?
Какова вероятность того, что хотя бы один мяч белый?

In [5]:
def combination(a,b):
    return np.math.factorial(b)/(np.math.factorial(a)*np.math.factorial(b-a))

In [6]:
P1 = (combination(2,7)/combination(2,10))*(combination(2,9)/combination(2,11))
P1

0.3054545454545455

Ответ: вероятность того, что все мячи белые - 30.54%

In [7]:
P2 = (combination(2,7)*combination(2,2)+combination(1,7)*combination(1,3)*combination(1,9)*combination(1,2)+
      combination(2,3)*combination(2,9))/(combination(2,10)*combination(2,11))

P2

0.20484848484848484

Ответ: вероятность того, что ровно два мяча белые - 20.48%

In [8]:
P3 = 1 - (combination(2,3)*combination(2,2))/(combination(2,10)*combination(2,11))
P3

0.9987878787878788

Ответ: вероятность того, что хотя бы один мяч белый - 99.88%